cirq/ops/gate_operation.py

```diff
@@ -187,6 +187,66 @@ def _qasm_(self, args: protocols.QasmArgs) -> Optional[str]:
                               default=None)
 
 
+@value.value_equality(approximate=True)
+class IdentityOperation(raw_types.Operation):
+    """An application of the identity gate to a sequence of qubits."""
+
+    def __init__(self, qubits: Sequence[raw_types.Qid]) -> None:
+        """
+        Args:
+            qubits: The qubits to operate on.
+        """
+        if len(qubits) == 0:
+            raise ValueError(
+                'Applied an identity gate to an empty set of qubits.')
+
+        if any([not isinstance(qubit, raw_types.Qid) for qubit in qubits]):
+            raise ValueError('Gate was called with type different than Qid.')
```

Comment:
`raise ValueError('Gave non-Qid objects to IdentityOperation: {!r}'.format(qubits))`